commands/azureCommands/acr-logs-utils/tableDataManager.ts

```diff
@@ -0,0 +1,136 @@
+
+    constructor(client: ContainerRegistryManagementClient, registry: Registry, resourceGroup: string) {
+        this.registry = registry;
+        this.resourceGroup = resourceGroup;
+        this.client = client;
+        this.logs = [];
+        this.links = [];
+    }
+    /** Acquires Links from an item number corresponding to the index of the corresponding log, caches
+     * logs in order to avoid unecessary requests if opened multiple times.
+     */
+    public async getLink(itemNumber: number): Promise<string> {
+        if (itemNumber >= this.links.length) {
+            throw new Error('Log for which the link was requested has not been added');
+        }
+
+        if (this.links[itemNumber].url) {
+            return this.links[itemNumber].url;
+        }
+
+        //If user is simply clicking many times impatiently it makes sense to only have one request at once
+        if (this.links[itemNumber].requesting) { return 'requesting' }
+
+        this.links[itemNumber].requesting = true;
+        const temp: RunGetLogResult = await this.client.runs.getLogSasUrl(this.resourceGroup, this.registry.name, this.logs[itemNumber].runId);
+        this.links[itemNumber].url = temp.logLink;
+        this.links[itemNumber].requesting = false;
+        return this.links[itemNumber].url
+    }
+
+    //contains(TaskName, 'testTask')
+    //`TaskName eq 'testTask'
+    //
+    /** Loads logs from azure
+     * @param loadNext Determines if the next page of logs should be loaded, will throw an error if there are no more logs to load
+     * @param removeOld Cleans preexisting information on links and logs imediately before new requests, if loadNext is specified
+     * the next page of logs will be saved and all preexisting data will be deleted.
+     * @param filter Specifies a filter for log items, if run Id is specified this will take precedence
+     */
+    public async loadLogs(loadNext: boolean, removeOld?: boolean, filter?: Filter): Promise<void> {
+        let runListResult: RunListResult;
+        let options: {
+            filter?: string,
+            top?: number,
+            customHeaders?: {
+                [headerName: string]: string;
+            };
+        } = {};
+        if (filter && Object.keys(filter).length) {
+            if (!filter.runId) {
+                options.filter = await this.parseFilter(filter);
+                if (filter.image) { options.top = 1; }
+                runListResult = await this.client.runs.list(this.resourceGroup, this.registry.name, options);
+            } else {
+                runListResult = [];
+                runListResult.push(await this.client.runs.get(this.resourceGroup, this.registry.name, filter.runId));
+            }
+        } else {
+            if (loadNext) {
+                if (this.nextLink) {
+                    runListResult = await this.client.runs.listNext(this.nextLink);
+                } else {
+                    throw new Error('No more logs to show');
```

Comment:
Is it not ok to throw new errors? Or is it because there is no try handling this method when used in the other functions?

Reply:
Given that this is in an HTML context, I'm not sure that's anything's actually handling it.  You'll probably have to play with it and decide the right thing to do (not sure showing a message is required anyway, depending on the UI).

Reply:
loadLogs can be, in fact, call by the Table when clicking in open logs, but it is also called when right clicking to see any type of logs. and loading them to the logData type defined in this file. 
So I think, that we should keep the throw, and handle catch the error in the setupIncomingListeners method. It can get an error thrown from other functions anyway.
Will this be ok?